data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/Evaluator.java

```diff
@@ -0,0 +1,11 @@
+package org.opensearch.dataprepper.expression;
+
+/**
+ * @since 1.3
+ * Placeholder interface for evaluators used by a {@link StatementEvaluator} implementation
+ * @param <ParsedData> parsed data type
+ * @param <Context> Context data type
+ */
+interface Evaluator<ParsedData, Context> {
+    Object evaluate(final ParsedData parsedData, final Context context) throws ClassCastException;
```

Comment:
Can you provide an example of an evaluator ? We should consider making the return type generic to avoid explicit casting from the caller.

`T evaluate(final ParsedData parsedData, final Context context) throws ClassCastException;`

Reply:
I'm planning for a ParseTreeEvaluator implementation that will implement `Evaluator<ParseTree, Event>`. The evaluator could parse a statement like `5` to an int, or `true` to a boolean. It would be the ConditionalExpressionEvaluators job to check the return type of ParseTreeEvaluator.evaluate or throw an exception.

Quick summery of class and responsibilitys:
`Parser` String -> ParseTree
`Evaluator` ParseTree -> Java Value
`StatementExpressionEvaluator` Parse -> Evaluator -> check result is expected type and return

I can make the return type of evaluator explicit to narrow the flexibility of the Evaluator for now if thats the direction we should go.